controller/client/client.go

```diff
@@ -492,7 +492,7 @@ func (c *Client) StreamDeployment(d *ct.Deployment, output chan *ct.DeploymentEv
 	}, appEvents)
 }
 
-func (c *Client) DeployAppRelease(appID, releaseID string) error {
+func (c *Client) DeployAppRelease(appID, releaseID string, logf func(*ct.DeploymentEvent)) error {
```

Comment:
@titanous I intend to re-work deployments to make them jobs after merging #2744 (which will aid in fixing a bunch of other deployment related issues), so I wonder if we should either 1) go with the current `logf` solution or 2) defer this change to when deployments are re-worked?